DependencyInjection/ReceiverLocatorPass.php

```diff
@@ -12,16 +12,15 @@ class ReceiverLocatorPass implements CompilerPassInterface
      */
     public function process(ContainerBuilder $container)
     {
-        if ($container->hasDefinition('messenger-monitor.receiver-locator')) {
+        if ($container->hasDefinition('karo-io.messenger_monitor.receiver_locator')) {
             if ($container->hasDefinition('console.command.messenger_consume_messages')) {
-
                 // steal configurations already done by the MessengerPass so we dont have to duplicate the work
                 // as approved by @ryanweaver with the "I've seen Nicolas do worse" certificate
                 $receiverLocatorDefinition = $container->getDefinition('karo-io.messenger_monitor.receiver_locator');
 
                 $consumeCommandDefinition = $container->getDefinition('console.command.messenger_consume_messages');
                 $names = $consumeCommandDefinition->getArgument(4);
-                $receiverLocatorDefinition->addArgument($names);
+                $receiverLocatorDefinition->replaceArgument(1, $names);
```

Comment:
can't we use `$container->getExtensionConfig('framework')` and merge all `framework.messenger.transports` we should find in order to remove this hack?

Reply:
I'm not sure. I'm not sure if I've ever seen that before. In theory, because this is a compiler pass, that should be available (I think?). But we would also be responsible for merging the array of arrays (from multiple config files). I'm not sure it's less of a hack.

Reply:
at least we would not rely on arguments of other services and their order

Reply:
up!